tools/ci_build/github/apple/build_host_protoc.sh

```diff
@@ -22,9 +22,11 @@ pushd .
 mkdir -p "$PROTOC_BUILD_PATH"
 cd "$PROTOC_BUILD_PATH"
 DEP_FILE_PATH="$ORT_REPO_ROOT/cmake/deps.txt"
+PATCH_FILE_PATH="$ORT_REPO_ROOT/cmake/patches/protobuf/protobuf_cmake.patch"
```

Comment:
I think as long as our generated protobuf files don't contain this `mach_absolute_time()` call, we don't need to patch the source used to build protoc.

the other ios pipelines which don't call build_host_protoc.sh use a downloaded protoc.

Reply:
Will look at removing build_host_protoc.sh in follow-up PR to clarify the setup a little more.